ct/alpine-vaultwarden.sh

```diff
@@ -73,7 +73,22 @@ function update_script() {
       exit
       ;;
     2)
-      whiptail --title "ADMIN TOKEN" --msgbox "$(cat /etc/conf.d/vaultwarden | grep ADMIN_TOKEN | awk '{print substr($2, 13) }')" 7 68
+      if NEWTOKEN=$(whiptail --passwordbox "Setup your ADMIN_TOKEN (make it strong)" 10 58 3>&1 1>&2 2>&3); then
+        if [[ -z "$NEWTOKEN" ]]; then exit-script; fi      
+        ADMINTOKEN=$(echo -n ${NEWTOKEN} | argon2 "$(openssl rand -base64 32)" -e -id -k 19456 -t 2 -p 1)
+        if [[ -f /var/lib/vaultwarden/config.json ]]; then 
```

Comment:
Why this? There is no file `/var/lib/vaultwarden/config.json`

Reply:
that one I found out during my testing with those 9 files :)

If you setup vaultwarden with the script the first time then the /etc/conf.d/vaultwarden will be read (I mean it will be read all the time if you NEVER changed anything inside the webui/admin panel).

If you ever changed ONE setting on the admin page, this file will be created and then this one is the mainfile.
Thats why we have to change the ADMIN_TOKEN there if anyone changed his settings :)

Small question before I start the part with "no setup during first installation": The whiptail menu during the update script is ok?

I can do a read command like you did on influxdb script for vaultwarden installation ofc... just thought that whiptail looks best and feels more "complete" when its done during the installation process?

Reply:
> 

Ok new commit where I removed the admin token during first installation.
Added the argon2 check during the update process (like you did with newt package)... so if a user starts this script and is using an "older" version of the alpine installation it will be added now
There was a missing "," at the end of the admin_token line in `config.json`, also fixed

Reply:
I like the way you're doing the whiptail in the update portion.
Can `/etc/conf.d/vaultwarden` be eliminated in favor of  `/var/lib/vaultwarden/config.json` during installation?

Reply:
> I like the way you're doing the whiptail in the update portion. Can `/etc/conf.d/vaultwarden` be eliminated in favor of `/var/lib/vaultwarden/config.json` during installation?

I tested it but it seems like it needs at least all the 4 lines except the admintoken line... I will test further if it is mandatory if we get those lines inside the config.json as well but I guess no...

tho the only thing we can get rid of is the admin token from this, don't know if it is worth it to make an other if/then loop to take care of this?

Reply:
have an Idea on making the script look cleaner on the update script... give me 5 more minutes please :)

Reply:
I doubt a symbolic link would work either

Reply:
the user can change stuff on the admin-page and the 2 files are not the same from content side.... a symbolic link won't help here (if I understand what you mean with that... ln command right?)